### DISCUSSION TITLE
Present the calculated values in the IPF plot

### DISCUSSION THREAD

**Discussion by ksafaei** [mtex-toolbox/mtex#737] [Ask Anything]
Hello,

My name is Keyvan, a Ph.D. student in mechanical engineering and I'm very new to MTex. I'm working with NiTi alloy (BCC structure) showing the crystallographic orientation dependency for strain/stress behavior. I can calculate the strain/stress value of any crystallographic orientation. Now, I want to present those values in an Inverse Pole Figure (IPF). So, How can I generate different orientations and show the calculated strain/stress in the IPF? An example of IPF plots has been attached.

Thanks in advance,
Keyvan

![Capture111](https://user-images.githubusercontent.com/78031590/106291588-d14b2780-6219-11eb-8c38-17ed547c04c7.JPG)

**Comment by kilir** [comment]
Hi,
you can generate for example a grid of orientations by
```
cs = crystalSymmetry('m3m')
ori = equispacedSO3Grid(cs)
```
For plotting an ipf, see here
https://mtex-toolbox.github.io/orientation.plotIPDF.html

Cheers,
Rüdiger

**Reply by ksafaei** [reply]
Hi Rudiger,

Thanks for your prompt reply. I was trying what you suggested. When the orientations are created, how I can get the (hkl) of them. I saw that it creates 4958 orientations and it has phi1, phi, phi2, and a, b,c, d. So, How can I extract the (hkl) to use for my calculation?

And the 2nd question: After I got the RSSF value for each orientation, how I can plot it in IPF. what specific Plot command and what are the variables?

Thank you very much for your time.
Keyvan

**Reply by kilir** [reply]
Hi Keyvan,

1) you have to describe a little more in detail what you mean with "extract the (hkl)".

2) for plotting inverse polefigures from orientations, contoured for some data (e.g. your RSSF), see the link in my first answer.

Cheers,
Rüdiger

**Reply by ksafaei** [reply]
Many thanks for your response. The ori. command that you suggested generates over 4000 different crystal orientations. I'm trying to call each plane orientation(for example (0 2 4) ) and calculate the RSSF for it and then plot the calculated results in IPF. I used these codes "r=vector3d.Z;  h=inv(ori)*r;" ( to call generate (hkl ) and then do some calculation on it and save the results in a matrix, I guess I could find the answer to my first question. Now I have a matrix with a size of 4000x1 that has the same column size as the orientation matrix. Now I want to have a contour IPF plot that shows the calculated values. I use " plotIPDF(ori,data,[r1,r2,r3])" for plotting, but it's not the one that I'm looking for. I've attached the image of the plot. So, How can I have a plot similar to those that I previously shared in my first massage?

Thanks again for your time!
Keyvan

![image](https://user-images.githubusercontent.com/78031590/107254129-ffdcb580-6a04-11eb-8c4f-06285158bd76.png)

**Comment by kilir** [comment]
Hi,

if I understand you correctly, what you do so far, is that you compute the direction (! not orientation) of a plane of type Miller using `inv(orientation)*r` and compute from this a scalar value, which is the one you use as `data` in plotIPDF. Then Mtex should contour those values in the plot of Millers from `inv(orientation)*r`.

If you do not like your result, could it be that you do not necessarily consider all symmetrically equivalent solutions for your crystal directions/planes?

Cheers,
Rüdiger

**Reply by ksafaei** [reply]
Hi,

Thanks, Rudiger. I was able to plot the results. I have some issues with the property of the plot and I will ask you if I can't find the solution.
Thank you again for your helpful response.
![image](https://user-images.githubusercontent.com/78031590/107542683-6f35df00-6b96-11eb-9d56-b148ab097d68.png)

Best,
Keyvan
